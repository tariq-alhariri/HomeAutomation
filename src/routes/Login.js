global.self = global;
import React from 'react';
import {
    StyleSheet,
    View,
    Text,
    Image,
    TouchableOpacity,
    TouchableHighlight,
    TextInput,
    Button,
    Alert,
    KeyboardAvoidingView
} from 'react-native';
export default class Login extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: ''
        };
    }
    async Login() {
        try {
            let response = await fetch('https://home99.herokuapp.com/login', {
                method: 'POST',
                headers: {
                    Accept: 'application/json',
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    user: {
                        username: this.state.username,
                        password: this.state.password
                    }
                })
            });

            let res = await response.text();
            console.log(res);
            res = JSON.parse(res);
            if (res == 'done') {
                return this.props.changeV('Home');
                console.log('exist');
            } else {
                Alert.alert('username or password incorrect');
            }
        } catch (errors) {
            console.log('catch errors' + errors);
        }
    }
    render() {
        return (
            <KeyboardAvoidingView behavior="padding" style={styles.container}>
                <View style={styles.logoContainer}>
                    <Image
                        style={styles.logo}
                        source={require('./Smart.png')}
                    />
                </View>
                <Text style={styles.header}>
                    Welcome to the Home Automation System
                </Text>

                <View>
                    <TextInput
                        placeholder="username"
                        returnKeyType="next"
                        validators="required"
                        onSubmitEditing={()=> this.passwordInput.focus()}
                        onChangeText={value =>
                        this.setState({ username: value })}
                        placeholderTextColor="#800080"
                        style={styles.input}
                    />

                    <TextInput
                        placeholder="password"
                        secureTextEntry={true}
                        validators="required"
                        returnKeyType="go"
                        ref={(Input)=> this.passwordInput = Input}
                        onChangeText={value =>
                        this.setState({ password: value })}
                        placeholderTextColor="#800080"
                        style={styles.input}
                    />
                    <View style={{flex: 0, flexDirection: 'row',backgroundColor: '#87CEFA'}}>
                    <TouchableHighlight
                        style={styles.buttonRow}
                        onPress={() => this.Login()}
                    >
                        <Text style={styles.buttonText}>Submit</Text>
                    </TouchableHighlight>

                    <TouchableHighlight
                        style={styles.buttonRow}
                        onPress={() => this.props.changeV('Signup')}
                    >
                        <Text style={styles.buttonText}>Go to Signup</Text>
                    </TouchableHighlight>

                    </View>
                </View>
            </KeyboardAvoidingView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#87CEFA',
        alignItems: 'center',
        padding: 60,
        justifyContent: 'center'
    },
    
    logoContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        flexGrow: 1,
        paddingBottom : 30,
        marginBottom: 35
    },

    logo: {
        width: 200,
        height: 200
    },
    header: {
        color: '#fff',
        marginBottom:0,
        marginTop: 5,
        marginBottom: 30,
        textAlign: 'center',
        opacity: 0.8,
        fontWeight: '700',
        fontSize :20
    },
    input: {
        marginBottom: 20,   
        paddingHorizontal: 10,
        backgroundColor: '#E0FFFF',
        marginLeft: 85,
        textAlign: 'center',
        justifyContent: 'center',
        color: '#94336A',  
        height: 40,      
        width: 200
        
    },
    buttonContainer: {
        backgroundColor: '#94336A',
        paddingVertical: 15
    },
    buttonText: {
        marginTop: -5,
        textAlign: 'center',
        color: '#ffffff',
        fontWeight: '700'
    },
        buttonRow :{    
        backgroundColor: '#4F1335',
        paddingVertical: 15,
        alignItems: 'center',
        marginBottom: 100,
        marginLeft: 40,
        marginRight: 5,
        borderRadius:10,
        borderWidth: 1,
        borderColor: '#94336A',
        height: 40,
        width: 140
    }
});