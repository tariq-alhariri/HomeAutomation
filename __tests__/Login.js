import 'react-native';
import React from 'react';
import Login from '../src/routes/Login';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('Search snapshot test', () => {
  const component = renderer.create(<Login />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot();
})

