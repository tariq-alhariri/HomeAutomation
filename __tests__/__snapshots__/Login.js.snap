// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot test 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#87CEFA",
        "flex": 1,
        "justifyContent": "center",
        "padding": 60,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
        "marginBottom": 35,
        "paddingBottom": 30,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "700",
        "marginBottom": 30,
        "marginTop": 5,
        "opacity": 0.8,
        "textAlign": "center",
      }
    }
  >
    Welcome to the Home Automation System
  </Text>
  <View>
    <TextInput
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="username"
      placeholderTextColor="#800080"
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#E0FFFF",
          "color": "#94336A",
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginLeft": 85,
          "paddingHorizontal": 10,
          "textAlign": "center",
          "width": 200,
        }
      }
      validators="required"
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="password"
      placeholderTextColor="#800080"
      returnKeyType="go"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#E0FFFF",
          "color": "#94336A",
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginLeft": 85,
          "paddingHorizontal": 10,
          "textAlign": "center",
          "width": 200,
        }
      }
      validators="required"
    />
    <View
      style={
        Object {
          "backgroundColor": "#87CEFA",
          "flex": 0,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#4F1335",
              "borderColor": "#94336A",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 40,
              "marginBottom": 100,
              "marginLeft": 40,
              "marginRight": 5,
              "paddingVertical": 15,
              "width": 140,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontWeight": "700",
              "marginTop": -5,
              "textAlign": "center",
            }
          }
        >
          Submit
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#4F1335",
              "borderColor": "#94336A",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 40,
              "marginBottom": 100,
              "marginLeft": 40,
              "marginRight": 5,
              "paddingVertical": 15,
              "width": 140,
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontWeight": "700",
              "marginTop": -5,
              "textAlign": "center",
            }
          }
        >
          Go to Signup
        </Text>
      </View>
    </View>
  </View>
</View>
`;
